api/api/configuration.py

```diff
@@ -79,8 +79,14 @@ def read_config() -> Dict:
         "experimental_features": False
     }
 
-    with open(common.api_config_path) as f:
-        configuration = yaml.safe_load(f)
+    if os.path.exists(common.api_config_path):
+        try:
+            with open(common.api_config_path) as f:
+                configuration = yaml.safe_load(f)
+        except IOError:
+            raise APIException(2000, "Error reading 'config.yml' file")

```

Comment:
The python exception message should be added to the exception. This error gives no clue about what's happening.

Reply:
The error code is also wrong. 2000 is already reserved when reading parameters not expected in configuration file.